feat: update testing db instructions lesson (#27974)
TheOdinProject/curriculum: nodeJS/testing_express/testing_database_operations.md

Before diving in, you might want to consider whether or not the database operations you're testing even need to be tested in the first place. If you are reading and writing straight from a database using `pg` or some other db module, you might not really need to test that code. [pg's GitHub repo's tests directory](https://github.com/brianc/node-postgres/tree/master/packages/pg/test) (and presumably all other popular db modules) already has plenty of tests for all of its actions, so if you are just serving up a JSON API and all you're doing is leveraging functions from another module then those operations are already covered.

If your queries are complicated, you might justify adding unit tests to make sure that you are using them correctly and that the code you have written is doing what you intend it to, and if you are using any of your own code to do some filtering, sorting, or other manipulations of the data you will want to test that as well.  In the case of your own code, however, it would be better if you could pull those things out into their own modules, separate from your database operations so you can test them without messing with the database.

### Integration tests

> There are cases, however, when you are going to want to test things that touch your database. For example, we performed an integration test on the server in the [testing routes and controllers lesson](./testing_routes_and_controllers.md) using `supertest`.

Reviewer: I think a full URL would be better here. If dead-set on a relative URL, then replacing `_` for `-` does work, and you can even drop the `.md` as well. But having `_` breaks as our URLs use split with `-`.
```suggestion
There are cases, however, when you are going to want to test things that touch your database. For example, we performed an integration test on the server in the [Testing Routes and Controllers lesson](https://www.theodinproject.com/lessons/nodejs-testing-routes-and-controllers) using `supertest`.
```

Author: it should definitely be a full URL, I got lazy lol

Create a separate database for testing; prefix it with a `test_` for you to easily identify with. You might want to run a prisma migrate by swapping out the test database url. Afterwards, use a seed script on this database instance, or you could manually insert data in a `beforeAll` function in your test suite.

We'll leverage environment variables for your codebase to identify which database to use. First, add the test database in your `.env` file:

> ```text

Reviewer: We get syntax highlighting for `.env` with this :tada:
````suggestion
```properties
````

NODE_ENV=development
DATABASE_URL=postgresql://<user>:<password>@localhost:3306/inventory_application
TEST_DATABASE_URL=postgresql://<user>:<password>@localhost:3306/test_inventory_application
```

Next, setup appropriate npm scripts in your `package.json` file:

```json
{
  // other stuff
  "scripts": {
    "dev": "NODE_ENV=development && node app.js",
    "test": "NODE_ENV=test && jest"
  },
  // even more stuff
}
```

Based on the `NODE_ENV`, you can programmatically switch out database urls:

```javascript

> const databaseUrl = process.env.NODE_ENV === 'test' ? process.env.TEST_DATABASE_URL : process.env.DATABASE_URL;

Reviewer: Can format this way to prevent horizontal scrolling.

```suggestion
const databaseUrl = process.env.NODE_ENV === 'test'
  ? process.env.TEST_DATABASE_URL
  : process.env.DATABASE_URL;
```

const prisma = new PrismaClient({
  datasources: {
    db: {
      url: databaseUrl,
    },
  },
});

> 

Reviewer: Rogue blank line
```suggestion
```

```

> Viola, the setup is complete. Now shoo... go get testin'.

Reviewer: :violin: 
```suggestion
Voila, the setup is complete. Now shoo... go get testin'.
```

Reviewer: @01zulfi LMAO that commit message
